_rules/name-from-content-included-in-accessible-name-2ee8b8.md

```diff
@@ -0,0 +1,200 @@
+## Background
+
+- [Understanding Success Criterion 2.5.3: Label in Name](https://www.w3.org/WAI/WCAG21/Understanding/label-in-name.html)
+- [Technique G208: Including the text of the visible label as part of the accessible name](https://www.w3.org/WAI/WCAG21/Techniques/general/G208)
+- [Technique G211: Matching the accessible name to the visible label](https://www.w3.org/WAI/WCAG21/Techniques/general/G211)
+
+## Test cases
+
+### Passed
+
+#### Passed Example 1
+
+Visible label and accessible name matches when trailing white spaces are removed.
+
+```html
+<div role="link" aria-label="next page ">next page</div>
+```
+
+#### Passed Example 2
+
+Character insensitivity between visible label and accessible name.
+
+```html
+<div role="link" aria-label="Next Page">next page</div>
+```
+
+#### Passed Example 3
+
+Full visible label is included in the accessible name, though the accessible name contains more text than the visible label.
+
+```html
+<button name="link" aria-label="Next Page in the list">Next Page</button>
+```
+
+#### Passed Example 4
+
+Full visible label is included in the accessible name, though the accessible name contains more text than the visible label.
+
+```html
+<head>
+  <style>
+    .sr-only {
+      position:absolute;
+      left:-10000px;
+      top:auto;
+      width:1px;
+      height:1px;
+      overflow:hidden;}
+  </style>
+</head>
+
+<body>
+  <a href="home.html">Read more<span class="sr-only"> about our pricing</span></a>
+</body>
+```
+
+#### Passed Example 5
+
+Text nodes in name from content are not expressing anything in human language and do therefore not live up to the [WCAG definition of text](https://www.w3.org/TR/WCAG21/#dfn-text).
+
+```html
+<button aria-label="close">:-)</button>
+```
+
+#### Passed Example 6
+
+Visible label and accessible name matches when punctuation is removed.
+
+```html
+<div role="link" aria-label="Next page">Next page...</div>
+```
+
+#### Passed Example 7
+
+Visible label and accessible name matches when parts that do not express anything in human language, "X", and punctuation is removed.
+
+```html
+<div role="link" aria-label="Close and go back">Close (X) and go back</div>
+```
+
+### Failed
+
+#### Failed Example 1
+
+Visible label is not included in accessible name.
+
+```html
+<div role="link" aria-label="OK">Next</div>
+```
+
+#### Failed Example 2
+
+Not all of visible label is included in accessible name.
+
+```html
+<button name="link" aria-label="the full">The full label</button>
+```
+
+#### Failed Example 3
+
+The full visible label is split out across the accessible name and is therefore not [included](#included-characters) in it.
+
+```html
+<head>
+  <style>
+    .sr-only {
+      position:absolute;
+      left:-10000px;
+      top:auto;
+      width:1px;
+      height:1px;
+      overflow:hidden;}
+  </style>
+</head>
+
+<body>
+  <a href="home.html">Link <span class="sr-only">that opens a new tab</span> to the homepage</a>
+</body>
+```
+
+### Inapplicable 
```

Comment:
We need examples for other inapplicable cases:
- no accessible name;
- text content is visible but not accessible (hence not a WCAG label);
- text content is accessible but not visible (hence not a WCAG label).

The later two cases, preferably with the text content not included in the name. _e.g._ 
`<a href="w3.org" aria-label="Home page of the W3C"><span aria-hidden=true>Go to W3C</span></a>`
is **not** a violation of 2.5.3 because there is no label (in the WCAG sense of the term). Thus, it should also be inapplicable for this rule.